src/Sweep.cpp

```diff
@@ -961,102 +885,101 @@ void InfectSweep(double t, int run) // added run number as argument in order to
+				Person* InfectiousPerson = Hosts + InfectiousPersonIndex;	//// person
 
-				unsigned short int tc; //// time at which person becomes case (i.e. moves from infectious and asymptomatic to infectious and symptomatic).
-				/* Following line not 100% consistent with DoIncub. All severity time points (e.g. SARI time) are added to latent_time, not latent_time + ((int)(P.LatentToSymptDelay / P.ModelTimeStep))*/
-				tc = si->latent_time + ((int)(P.LatentToSymptDelay / P.ModelTimeStep)); //// time that person si/ci becomes case (symptomatic)...
-				if ((P.DoSymptoms) && (ts == tc)) //// ... if now is that time...
-					DoCase(ci, t, ts, tn);		  //// ... change infectious (but asymptomatic) person to infectious and symptomatic. If doing severity, this contains DoMild and DoILI.
+				unsigned short int CaseTime; //// time at which person becomes case (i.e. moves from infectious and asymptomatic to infectious and symptomatic).
+				CaseTime = InfectiousPerson->latent_time + ((int)(P.LatentToSymptDelay / P.ModelTimeStep)); //// time that person si/ci becomes case (symptomatic)...
+				if ((P.DoSymptoms) && (TimeStepNow == CaseTime)) //// ... if now is that time...
+					DoCase(InfectiousPersonIndex, t, TimeStepNow, ThreadNum);		  //// ... change infectious (but asymptomatic) person to infectious and symptomatic. If doing severity, this contains DoMild and DoILI.
 
-				if (P.DoSeverity)
+				if (P.DoSeverity) // Don't be tempted to if/else or switch following code as there are edge cases where e.g. SARI_time = recovery_or_death_time, and so they're not mutually exclusive.
 				{
-					if (ts >= si->SARI_time)		DoSARI					(ci, tn);	//// see if you can dispense with inequalities by initializing SARI_time, Critical_time etc. to USHRT_MAX
-					if (ts >= si->Critical_time)	DoCritical				(ci, tn);
-					if (ts >= si->Stepdown_time)	DoRecoveringFromCritical(ci, tn);
-					if (ts >= si->recovery_or_death_time)
+					if (TimeStepNow == InfectiousPerson->SARI_time)		DoSARI(InfectiousPersonIndex, ThreadNum);	//// see if you can dispense with inequalities by initializing SARI_time, Critical_time etc. to USHRT_MAX
+					if (TimeStepNow == InfectiousPerson->Critical_time)	DoCritical(InfectiousPersonIndex, ThreadNum);
+					if (TimeStepNow == InfectiousPerson->Stepdown_time)	DoRecoveringFromCritical(InfectiousPersonIndex, ThreadNum);
+					if (TimeStepNow == InfectiousPerson->recovery_or_death_time)
 					{
-						if (si->to_die)
-							DoDeath_FromCriticalorSARIorILI(ci, tn);
+						if (InfectiousPerson->to_die)
+							DoDeath_FromCriticalorSARIorILI	(InfectiousPersonIndex, ThreadNum);
 						else
-							DoRecover_FromSeverity(ci, tn);
+							DoRecover_FromSeverity(InfectiousPersonIndex, ThreadNum);
 					}
 				}
 
 				//Adding code to assign recovery or death when leaving the infectious class: ggilani - 22/10/14
-				if (ts >= si->recovery_or_death_time)
+				if (TimeStepNow == InfectiousPerson->recovery_or_death_time)
```

Comment:
Similar comment to eralier on >= changing into ==

Ok if you're sure that this gets called every timestep, and the right-hand-sides don't change so the event gets missed.